files/provision-all

```diff
@@ -31,6 +31,10 @@ sudo qubesctl --show-output --skip-dom0 --targets sd-small-bookworm-template sta
 # Reboot sd-log so it's ready to receive logs from other VMs about to be configured
 qvm-shutdown --wait sd-log && qvm-start sd-log
 
+echo ".........................................................................."
+echo "Enabling Whonix customizations"
+sudo qubesctl --show-output --skip-dom0 --targets whonix-gateway-17 state.highstate
```

Comment:
I would appreciate input from a salt guru like @rocodes or @deeplow that this is safe to run (and also the correct invocation).

Reply:
This is the correct target, as long as there's an already-created vm present by that name. It does make me want us to clone the whonix template and do all our own operations on the cloned template though, which I think you mentioned somewhere else Kunal, since `provision-all` isn't just run once but any time a migration happens.

As deeplow mentioned this will run whatever states are present/enabled on the machine for the vm by this name.  If a user created a custom .top and .sls file that also targets whonix-gateway-17, that would also be run.  (That's kind of just the symptom of the underlying "should we rely on system templates in an ongoing way or not?" question).

Reply:
We've gone back and forth on our whonix provisioning approach in sdw: we previously used `qvm.anon-whonix`, which are upstream tested Salt states that configure the latest whonix vms and set up an up to date disposable anon-whonix vm. Now we invoke the template name directly, needing a little more attention when upgrading, to make sure we upgrade everywhere consistently. (It also doesn't change the user's Tor VMs, which is good or bad depending on how heavy-handed we think we should be).

I kind of think we should:
* run the upstream `qvm.anon-whonix` state again as part of our use of whonix (they describe it as a mandatory step - it configures [update policy information](https://www.whonix.org/wiki/Dev/Qubes#salt), in addition to setting up vms).
* Then clone the whonix gateway and name it whatever we want, and apply the custom configuration to that.

(I think what I'm suggesting can/should be a separate PR)

Reply:
Thanks, all.  @rocodes, if you're happy with your proposal coming as a follow-up change, I'll tackle that next, rather than blocking this.

Reply:
Filed as #1060.

Reply:
Thanks, sorry I missed your comment yesterday, but yes, not intended to block this PR, all followup work.